samples/subsys/mgmt/mcumgr/smp_svr/README.rst

```diff
@@ -1,117 +1,271 @@
+"OTA DFU", or Over-The-Air Device Firmware Upgrade.
+
+In order to do so you will need to build a second sample to verify that it is
+sent over the air, properly flashed into slot-1 and then swapped into slot-0
+by MCUboot.
+
+Build a second sample
+---------------------
+
+Perhaps the easiest sample to test with is the :file:`samples/hello_world`
+sample provided by Zephyr, documented in the :ref:`hello_world` section.
+
+Edit :file:`samples/hello_world/prj.conf` and enable the required MCUboot
+Kconfig option as described in :ref:`mcuboot` by adding the following line to it:
+
+.. code-block:: console
+
+   CONFIG_BOOTLOADER_MCUBOOT=y
+
+Then build the sample as usual (see :ref:`hello_world`).
+
+Sign the second sample
+----------------------
+
+Next you will need to sign the sample just like you did for :file:`smp_svr`,
+since it needs to be loaded by MCUboot all the same.
+To do that follow the same instructions described in :ref:`smp_svr_sample_sign`,
+but this time you must use a :file:`.bin` image, since :file:`mcumgr` does not
+yet support :file:`.hex` files.
+
+Upload the image over BLE
+-------------------------
+
+Now we are ready to send or upload the image over BLE to the target remote
+device.
+
+.. code-block:: console
+
+   sudo mcumgr --conntype ble --connstring ctlr_name=hci0,peer_name='Zephyr' image upload signed.bin
+
+If all goes well the image will now be stored in slot-1, ready to be swapped
+into slot-0 and executed.
+
+List the images
+---------------
+
+We can now obtain a list of images (slot-0 and slot-1) present in the remote
+target device by issuing the following command:
+
+.. code-block:: console
+
+   sudo mcumgr --conntype ble --connstring ctlr_name=hci0,peer_name='Zephyr' image list
+
+This should print the status and hash values of each of the images present.
+
+Test the image
+--------------
+
+In order to instruct MCUboot to swap the images we need to test the image first,
+making sure it boots:
 
 .. code-block:: console
 
-    $ mcumgr --conntype ble --connstring peer_name=Zephyr echo hello
-    hello
+   sudo mcumgr --conntype ble --connstring ctlr_name=hci0,peer_name='Zephyr' image test <hash of slot-1 image>
+
+Now MCUBoot will swap the image on the next reset.
+
+Reset remotely
+--------------
+
+We can reset the device remotely to observe (use the console output) how
+MCUboot swaps the images:
+
+.. code-block:: console
+
+   sudo mcumgr --conntype ble --connstring ctlr_name=hci0,peer_name='Zephyr' reset
+
+Upon reset MCUboot will swap slot-0 and slot-1.
+
+Since the new image is actually the basic ``hello_world`` sample it doesn't
+contain neither SMP nor BLE functionality, so we cannot communicate with it
+using :file:`mcumgr`. Instead simply reset the board manually to force MCUboot
+to revert (i.e. swap back the images) due to the fact that the new image has
+not been confirmed.
+
+If you however upload an image based on ``smp_svr`` or any other BLE and SMP
```

Comment:
Change to:

    If you had instead built and uploaded a new image based on ``smp_svr``
    (or another BLE and SMP enabled sample), you could confirm the
    new image and make the swap permanent by using this command: